src/context/AlertContext.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+
+import Alert from '@/components/UI/alert'
+
+export interface AlertContentProps {
+  title?: string
+  text: string
+  variant: AlertVariant
+  position?: 'top' | 'bottom'
+  confirmFunction?: () => void
+  cancelFunction?: () => void
+  showFor?: number
+}
+
+export enum AlertVariant {
+  info,
+  security,
+  critical,
+  comment
+}
+
+interface AlertContextProps {
+  setAlert: (content: AlertContentProps) => void
+  clearAlert: () => void
+  visible: boolean
+}
+
+const AlertContext = createContext<AlertContextProps>({
+  setAlert: () => {
+    return
+  },
+  clearAlert: () => {
+    return
+  },
+  visible: false
+})
+
+export const useAlert = () => useContext(AlertContext)
+
+export const AlertContextProvider = ({
+  children
+}: {
+  children: React.ReactNode
+}) => {
+  const [visible, setVisible] = useState(false)
+  const [content, setContent] = useState<AlertContentProps>({
+    title: '',
+    text: '',
+    position: 'top',
+    confirmFunction: undefined, // function to execute on confirm, enables the confirm function
+    cancelFunction: undefined,
+    showFor: 5000, // ms alert will stay open, set to -1 to leave open until button click
+    variant: AlertVariant.info
+  })
+
+  const setAlert = (content: AlertContentProps) => {
+    setContent({
+      title: content.title ?? '',
+      text: content.text,
+      variant: content.variant,
+      position: content.position ?? 'top',
+      confirmFunction: content.confirmFunction ?? undefined,
+      cancelFunction: content.cancelFunction ?? undefined,
+      showFor: content.showFor ?? 5000
+    })
+  }
+
+  const clearAlert = useCallback(() => {
+    setVisible(false)
+    setContent({
+      text: '',
+      title: '',
+      position: 'top',
+      confirmFunction: undefined,
+      cancelFunction: undefined,
+      showFor: 5000,
+      variant: AlertVariant.info
+    })
+  }, [])
```

Comment:
Originally was planning to use another timer to animate between alerts rather than it jumping and replacing, I have removed as will only add back in if this functionality is added.

Reply:
Removed this also, I was trying to prevent the alert from updating on setVisible but I believe it's been fixed via changes to the context.